jkube-kit/enricher/specific/src/test/java/org/eclipse/jkube/kit/enricher/specific/ServiceDiscoveryEnricherTest.java

```diff
@@ -69,51 +71,42 @@ public void setUp() {
     }
 
     @Test
-    public void testDiscoveryLabel() {
+    void discoveryLabel() {
         enricher.addAnnotations(builder);
         String value = builder.buildMetadata().getLabels().get("discovery.3scale.net");
         assertThat(value).isEqualTo(discoverable);
     }
 
     @Test
-    public void testDescriptionPathAnnotation() {
-        enricher.addAnnotations(builder);
-        assertAnnotation(descriptionPath, "description-path");
-    }
-
-    @Test
-    public void testDiscoveryVersionAnnotation() {
-        enricher.addAnnotations(builder);
-        assertAnnotation(discoveryVersion, "discovery-version");
-    }
-
-    @Test
-    public void testDefaultDiscoveryVersionAnnotation() {
+    void defaultDiscoveryVersionAnnotation() {
         context.getConfiguration().getProcessorConfig().getConfig().get("jkube-service-discovery").remove("discoveryVersion");
         enricher.addAnnotations(builder);
-        assertAnnotation("v1", "discovery-version");
+        Map<String, String> annotations = builder.buildMetadata().getAnnotations();
+        assertThat(annotations)
+            .containsEntry("discovery.3scale.net/discovery-version", "v1");
     }
 
-    @Test
-    public void testPathAnnotation() {
+    @DisplayName("add annotations to service")
+    @ParameterizedTest(name = "''{0}'' annotation should be ''{1}''")
+    @MethodSource("annotations")
+    void addAnnotations(String annotation, String expectedValue) {
         enricher.addAnnotations(builder);
-        assertAnnotation("/rest-3scale", "path");
-    }
-
-    @Test
-    public void testPortAnnotation() {
-        enricher.addAnnotations(builder);
-        assertAnnotation(port, "port");
+        Map<String, String> annotations = builder.buildMetadata().getAnnotations();
+        assertThat(annotations)
+            .containsEntry("discovery.3scale.net/" + annotation, expectedValue);
     }
 
-    @Test
-    public void testSchemeAnnotation() {
-        enricher.addAnnotations(builder);
-        assertAnnotation(scheme, "scheme");
+    static Stream<Arguments> annotations() {
+      return Stream.of(
+          Arguments.of("description-path", descriptionPath),
+          Arguments.of("discovery-version", discoveryVersion),
+          Arguments.of("path", "/rest-3scale"),
+          Arguments.of("port", port),
+          Arguments.of("scheme", scheme));
```

Comment:
I'm sorry for not being clear in my first comment. What I meant was that you don't need to have parameterized test here. You're providing a constant value as input to the test. Maybe it can be refactored like this:
```java
    @Test
    void addedAnnotations() {
        // Given + When
        enricher.addAnnotations(builder);

        // Then
        assertThat(builder.buildMetadata().getAnnotations())
            .containsEntry("discovery.3scale.net/description-path", descriptionPath)
            .containsEntry("discovery.3scale.net/discovery-version", discoveryVersion)
            .containsEntry("discovery.3scale.net/path", "/rest-3scale")
            .containsEntry("discovery.3scale.net/port", port)
            .containsEntry("discovery.3scale.net/scheme", scheme);
        // OR
        for (Map.Entry<String, String> entry : Arrays.asList(
            new AbstractMap.SimpleEntry<>("description-path", descriptionPath),
            new AbstractMap.SimpleEntry<>("discovery-version", discoveryVersion),
            new AbstractMap.SimpleEntry<>("path", "/rest-3scale"),
            new AbstractMap.SimpleEntry<>("port", port),
            new AbstractMap.SimpleEntry<>("scheme", scheme))) {
           assertAnnotation(entry.getValue(), entry.getKey());
        }
    }
```

Reply:
Got it!